core/runtime/register_trt_op.cpp

```diff
@@ -10,8 +10,92 @@ namespace trtorch {
 namespace core {
 namespace runtime {
 
+// Checks if the context switch requred for device ID
+bool is_switch_required(const CudaDevice& curr_device, const CudaDevice& conf_device) {
+  // If SM capability is not the same as configured then switch
+  if ((curr_device.major != conf_device.major) || (curr_device.minor != conf_device.minor)) {
+    LOG_WARNING("Configured SM capability does not match with current device ID. Switching context");
+    return true;
+  }
+
+  // GPU case
+  if (conf_device.device_type == nvinfer1::DeviceType::kGPU) {
+    if (curr_device.device_name != conf_device.device_name) {
+      LOG_WARNING(
+          "TRTEngine compiled for " << conf_device.device_name << " but current CUDA device is "
+                                    << curr_device.device_name << ". Switching the device context");
+      return true;
+    }
+  }
+
+  if (curr_device.id != conf_device.id) {
+    LOG_WARNING(
+        "Configured Device ID: " << conf_device.id << " is different that current device ID: " << curr_device.id
+                                 << ". Switching context");
+    return true;
+  }
+
+  return false;
+}
+
+int select_cuda_device(const CudaDevice& conf_device) {
+  int device_id = 0;
+  int num_devices = 0;
+  // SM Compute capability <major,minor> pair
+  std::unordered_map<std::string, std::string> dla_supported_SM;
+
+  // Xavier SM Compute Capability
+  dla_supported_SM.insert(std::make_pair("7.2", "Xavier"));
```

Comment:
Since Volta architecture also shares the SM capability. If we don't add Xavier/Orin to it will cause errors for dGPUs based on compute capability 7.2  such as Titan V cards.

I cannot think of any other way around this. For Orin, we will have to add that to the list as well.

Reply:
Can you make a static DLA supported SM map then like this: https://github.com/NVIDIA/TRTorch/blob/master/core/conversion/conversion_ignorelist.cpp

Reply:
Done